File: linkedList/queue.js
class QUEUE {
  constructor(){
    this.head = null;
    this.tail = null;
  }

  push(val){
    var newNode = new Node(val);
    if(!this.head){
      this.head = newNode;
      this.tail = newNode;
    }

  }


}

File: linkedList/queue.cpp
class Node {

	private:
		int val;
		Node* next;

	public:
		Node(int val){
			this->val = val;
			this->next = NULL;
		}

};

class Queue {

	public:
	
		Queue(){
			this->head = NULL;
			this->tail = NULL;
		}


	private:
		Node* head;
		Node* tail;

};